### ISSUE TITLE
Auto Import adds extra comma to import statements with trailing commas

### ISSUE THREAD

**Issue by cadin** [microsoft/TypeScript#40561]
*TS Template added by @mjbvz*

**TypeScript Version**: 4.0.2

**Search Terms**
- auto import

---

- VSCode Version: 1.49.0
- OS Version: macOS 10.15.6

Steps to Reproduce:

1. Import multiple files from the same package, listing files on separate lines with trailing commas.
Eg:
```
import {  
    Foo,  
    Bar,  
    Baz,  
} from "myPackage" 
```
2. Invoke an auto import from the same package by typing it's name (eg: "Qux")

**Results:**
The new file is added to the import statement with an extra comma:
```
import {  
    Foo,  
    Bar,  
    Baz,, Qux  
} from "myPackage" 
```

**Expected Results:**
The import should be added on a new line with trailing comma:
```
import {  
    Foo,  
    Bar,  
    Baz,
    Qux,  
} from "myPackage" 
```
The proper behavior occurs in version 1.48.2

Does this issue occur when all extensions are disabled?: Yes

**Comment by mjbvz** [comment]
Self contained example:

`x.ts`
```ts
export const a = 123;
export const b = 123;
export const c = 123;
```

`index.ts`
```ts
import {
    b,
    a,
} from './x'

console.log(c) // trigger import here
```

**Comment by DanielRosenwasser** [comment]
@andrewbranch does #40303 fix this for the 4.0 branch?

**Comment by andrewbranch** [comment]
Yes

**Comment by andrewbranch** [comment]
Well, I’m confused as to how this was happening on the nightly for the 11th. I’ll investigate, but this seems to be the exact same issue.

**Comment by andrewbranch** [comment]
Yeah, I can't reproduce this with the nightly mentioned or with current master. @mjbvz was your repro using a nightly TS build?

**Comment by swashata** [comment]
Hi,

My TS version is 4.0.2 and I can confirm this happening.

<img width="542" alt="Screen Shot 2020-09-16 at 11 17 06 AM" src="https://user-images.githubusercontent.com/1623381/93297102-43444680-f80e-11ea-83a9-26c769070743.png">

**Comment by mjbvz** [comment]
@andrewbranch Actually this looks fixed in nightly; I must have tested it incorrectly. Can we make sure this fix is included in the next 4.0 recovery?

**Comment by Shaker-Hamdi** [comment]
I have the same issue.

**Comment by andrewbranch** [comment]
@mjbvz it’s already in the release branch: https://github.com/microsoft/TypeScript/pull/40303

**Comment by leidegre** [comment]
> @mjbvz it’s already in the release branch: #40303

OK, what does that mean? Will it get fixed in the next minor of TypeScript?

**Comment by andrewbranch** [comment]
It will be in the next patch release, 4.0.3.

**Comment by tonix-tuft** [comment]
I am still facing this issue too, how can I fix it in my VS Code? -> https://github.com/microsoft/vscode/issues/107057

**Comment by mlewando** [comment]
I still have this problem :(

I installed globally the newest typescript (4.1.3) and set `typescript.tsdk` to my version. I have VS Code 1.52.1. Do I need to do anything more to have this fixed?

**Comment by andrewbranch** [comment]
@mlewando does the TS version in the bottom-right match what you set? 

<img width="205" alt="image" src="https://user-images.githubusercontent.com/3277153/104633309-d02fcd00-5653-11eb-83f9-551ff30defb6.png">

If so, please post exact steps for how to reproduce.

**Comment by mlewando** [comment]
Hi @andrewbranch I have a JavaScript project. I think that VS Code uses typescript under the hood anyway to autocomplete imports, but I don't have typescript version in my status bar:
![image](https://user-images.githubusercontent.com/3727118/104658398-fccefd80-56c2-11eb-9eed-1f7dc3c8ed54.png)

the exact steps to reproduce for not is rather "it happens from time to time", but if it is (or may be) related, I will try to identify the exact reproduction.

**Comment by andrewbranch** [comment]
In that case, when you have a JS file open, open the command palette and search for "Select TypeScript Version." Select it and make sure 4.1.3 is selected:

<img width="611" alt="image" src="https://user-images.githubusercontent.com/3277153/104658608-0da65180-5678-11eb-83e3-f3c09b88c7a6.png">

**Comment by mlewando** [comment]
it is 4.1.3
![image](https://user-images.githubusercontent.com/3727118/104658825-df4e6380-56c3-11eb-9d5c-bfb1faa5333e.png)

so, I'll try to identify when exactly it happens and hopefully tomorrow I'll post working reproduction

**Comment by andrewbranch** [comment]
Thanks! The text of all your imports before and after the auto import should be enough to reproduce.

**Comment by mlewando** [comment]
So, I created a simple reproduction without a problem, and then I tried to reproduce it on the simple example with all extensions disabled - I failed. So I started to disable extensions until I identified the problematic one: exports autocomplete (https://github.com/capaj/vscode-exports-autocomplete)
This extension is part of React Food Truck, but it became deprecated some time ago, which is even noted in https://github.com/burkeholland/react-food-truck/issues/6.

To sum up: for me, problem is solved :) thank you for attention :)
